x-pack/plugins/security_solution/server/endpoint/services/artifacts/manifest_manager/manifest_manager.ts

```diff
@@ -835,4 +867,241 @@ export class ManifestManager {
+    if (!globalUnifiedManifestSO?.saved_objects?.length) {
+      this.logger.warn('No Global Unified Manifest found to bump version');
+      return;
+    }
+    const globalUnifiedManifest = globalUnifiedManifestSO.saved_objects[0];
+
+    const newSemanticVersion =
+      this.setNewSemanticVersion(globalUnifiedManifest.attributes.semanticVersion) || '1.0.0';
+    await this.getUnifiedManifestClient().updateUnifiedManifest({
+      ...globalUnifiedManifest.attributes,
+      id: globalUnifiedManifest.id,
+      semanticVersion: newSemanticVersion,
+    });
+  }
+
+  public async commitUnified(manifestSo: InternalManifestSchema): Promise<void> {
+    const existingUnifiedManifestsSo = await this.getAllUnifiedManifestsSOFromCache();
+
+    const unifiedManifestSO = this.transformLegacyManifestSOtoUnifiedManifestSO(
+      manifestSo,
+      existingUnifiedManifestsSo
+    );
+
+    const { unifiedManifestsToUpdate, unifiedManifestsToCreate, unifiedManifestsToDelete } =
+      this.prepareUnifiedManifestsSOUpdates(unifiedManifestSO, existingUnifiedManifestsSo);
+
+    if (unifiedManifestsToCreate.length) {
+      await asyncForEach(chunk(unifiedManifestsToCreate, 100), async (unifiedManifestsBatch) => {
+        await this.getUnifiedManifestClient().createUnifiedManifests(unifiedManifestsBatch);
+      });
+      this.logger.debug(`Created ${unifiedManifestsToCreate.length} unified manifests`);
+    }
+
+    if (unifiedManifestsToUpdate.length) {
+      await asyncForEach(chunk(unifiedManifestsToUpdate, 100), async (unifiedManifestsBatch) => {
+        await this.getUnifiedManifestClient().updateUnifiedManifests(unifiedManifestsBatch);
+      });
+
+      this.logger.debug(`Updated ${unifiedManifestsToUpdate.length} unified manifests`);
+    }
+
+    if (unifiedManifestsToDelete.length) {
+      await asyncForEach(chunk(unifiedManifestsToDelete, 100), async (unifiedManifestsBatch) => {
+        await this.getUnifiedManifestClient().deleteUnifiedManifestByIds(unifiedManifestsBatch);
+      });
+
+      this.logger.debug(`Deleted ${unifiedManifestsToDelete.length} unified manifests`);
+    }
+
+    if (
+      unifiedManifestsToCreate.length ||
+      unifiedManifestsToUpdate.length ||
+      unifiedManifestsToDelete.length
+    ) {
+      // If global manifest is not in the list of manifests to create or update, we need to bump its version
+      // We use it to set schemaVersion of the legacy manifest we are going to create so that its being picked up when populating agent policy
+      const hasGlobalManifest = [...unifiedManifestsToCreate, ...unifiedManifestsToUpdate].some(
+        (manifest) => manifest.policyId === '.global'
+      );
+
+      if (!hasGlobalManifest || unifiedManifestsToDelete.length) {
+        await this.bumpGlobalUnifiedManifestVersion();
+      }
+      // Clear fetched Unified Manifests from cache since they are not up-to-date anymore
+      this.clearCachedUnifiedManifestsSO();
```

Comment:
yeah.   so any error along the way that breaks the flow of the code will result in this method from being called and thus stale data remaining in cache. Make sure you account for that and hopefully find a good solution